process/block/preprocess/scheduledTxsExecution.go

```diff
@@ -316,85 +366,98 @@ func (ste *scheduledTxsExecution) GetScheduledRootHashForHeader(
 
 // RollBackToBlock rolls back the scheduled txs execution handler to the given header
 func (ste *scheduledTxsExecution) RollBackToBlock(headerHash []byte) error {
-	scheduledRootHash, mapScheduledSCRs, err := ste.getScheduledRootHashAndSCRsForHeader(headerHash)
+	scheduledRootHash, mapScheduledSCRs, gasAndFees, err := ste.getScheduledRootHashSCRsAndGasForHeader(headerHash)
 	if err != nil {
 		return err
 	}
 
 	log.Debug("scheduledTxsExecution.RollBackToBlock", "header hash", headerHash, "scheduled root hash", scheduledRootHash, "num of scheduled scrs", len(mapScheduledSCRs))
 
-	ste.SetScheduledRootHashAndSCRs(scheduledRootHash, mapScheduledSCRs)
+	ste.SetScheduledRootHasSCRsAndGas(scheduledRootHash, mapScheduledSCRs, *gasAndFees)
 
 	return nil
 }
 
+// SaveStateIfNeeded saves the scheduled Txs Execution state for the given header hash, if there are scheduled TXs
 func (ste *scheduledTxsExecution) SaveStateIfNeeded(headerHash []byte) {
 	scheduledRootHash := ste.GetScheduledRootHash()
 	mapScheduledSCRs := ste.GetScheduledSCRs()
 	ste.mutScheduledTxs.RLock()
+	gasAndFees := ste.gasAndFees
 	numScheduledTxs := len(ste.scheduledTxs)
 	ste.mutScheduledTxs.RUnlock()
 	log.Debug("scheduledTxsExecution.SaveStateIfNeeded", "num of scheduled txs", numScheduledTxs)
 
 	if numScheduledTxs > 0 {
-		ste.SaveState(headerHash, scheduledRootHash, mapScheduledSCRs)
+		ste.SaveState(headerHash, scheduledRootHash, mapScheduledSCRs, gasAndFees)
```

Comment:
no need to save gasAndFees? Could there be a case with no scheduledTxs but somehow the gasAndFees changed?